reader-writer-lock/src/main/java/com/iluwatar/reader/writer/lock/App.java

```diff
@@ -0,0 +1,82 @@
+package com.iluwatar.reader.writer.lock;
+
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+
+/**
+ * Reader writer lock is a synchronization primitive that solves one of the readers–writers
+ * problems. An RW lock allows concurrent access for read-only operations, while write operations
+ * require exclusive access.
+ * <p>
+ * Below example use two mutexes to demonstrate the concurrent access of mutilple readers and
+ * writers.
+ * 
+ */
+public class App {
+
+  private static Random ran = new Random();
+
+  /**
+   * Program entry point
+   * 
+   * @param args command line args
+   */
+  public static void main(String[] args) {
+
+    ExecutorService es = Executors.newFixedThreadPool(1000);
+    ReaderWriterLock lock = new ReaderWriterLock();
+
+    AtomicInteger index = new AtomicInteger(0);
+    IntStream.range(0, 100).forEach(i -> {
+      Runnable task = null;
+      if (ran.nextFloat() <= 0.6) {
```

Comment:
Why use random numbers here? I think you can just allocate certain number of readers and writers.